### PR TITLE
Gulp fixes

### PR DESCRIPTION
**Pull request by maaarghk**
* Use gulp-load-plugins instead of manually loading in all the plugins
* Require that the clean task finishes before we move anything into /dist or we end up with a race condition.
* Add respository path to package.json so npm stops complaining
* Use glob instead of fs.readdir - only load js files.

### CODE REVIEW

File: build/tasks/build.js [hunk 1]
```diff
@@ -1,55 +1,54 @@
...
+        }
+    });
 
     // delete the output/tmp directories
     gulp.task('clean', function (callback) {
-        del([config.outputDir+'/*', 
-             config.tempDir+'/*'], callback);
+        del([config.outputDir+'/*',
+             config.tempDir+'/*',
+             '**/.DS_Store'], callback);
```
Comment (jrhouston) [inline build/tasks/build.js]: Why would we need to clean this directory, though?

File: gulpfile.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
-var fs = require('fs');
+var glob = require('glob');
 var gulp = require('gulp');
 var config = require('./build/config.js');
 
-fs.readdirSync('./build/tasks/').forEach(function (file) {
-    require('./build/tasks/'+file)(gulp, config);
+files = glob.sync('./build/tasks/*.js');
+files.forEach(function(file) {
```
Comment (jrhouston) [inline gulpfile.js]: Pedantry: can we leave a space between function and the opening parenthesis. Coding style, yo.

File: build/tasks/build.js (hunk 1 above)
Comment (maaarghk) [inline build/tasks/build.js]: The DS_Store clean isn't really required anymore since it uses glob now.

File: gulpfile.js (hunk 2 above)
Comment (maaarghk) [inline gulpfile.js]: Fire up a style guide at some point.

**Pull request merged**